File: 卡码/代码随想录/Go/回溯/03-子集/LeetCode_0090.go
// 90. 子集 II
package main

import "sort"

func subsetsWithDup(nums []int) [][]int {
	res:=make([][]int,0)
	path:=make([]int,0)
	var backtracking func(nums []int,index int)
	backtracking=func(nums []int,index int){
		tmp:=make([]int,len(path))
		copy(tmp,path)
		res=append(res,tmp)
		for i:=index;i<len(nums);i++ {
			if i>index&&nums[i]==nums[i-1] {
				continue
			}
			path=append(path,nums[i])
			backtracking(nums,i+1)
			path=path[:len(path)-1]
		}
	}
	sort.Ints(nums)
	backtracking(nums,0)
	return res
}

File: 卡码/代码随想录/JS/回溯/03-子集/LeetCode_0090.js
//90. 子集 II
const subsetsWithDup = function(nums) {
  const res=new Array();
  const path=new Array();
  const backtracking=function(nums,index){
    res.push(Array.from(path));
    for(let i=index;i<nums.length;i++){
      if(i>index&&nums[i]==nums[i-1]) continue;
      path.push(nums[i]);
      backtracking(nums,i+1);
      path.pop();
    }
  }
  nums.sort((a,b)=>a-b);
  backtracking(nums,0);
  return res;
};